### ISSUE TITLE
tensorflow: AutoGraph could not transform data

### ISSUE THREAD

**Issue by raposter** [tensorflow/tensorflow#48175]
Error reported in Python 3.9.2 tf-nightly:
"WARNING:tensorflow:AutoGraph could not transform <bound method Dense.call of <keras.layers.core.Dense object at 0x0000019783941EE0>> and will run it as-is.
Please report this to the TensorFlow team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output.
Cause: invalid syntax (tmp0qln7o16.py, line 48)
To silence this warning, decorate the function with @tf.autograph.experimental.do_not_convert"

This feature works in Tensorflow for Python 3.8 but not in tf-nightly for Python 3.9

**Comment by raposter** [comment]
from random import random
from numpy import array
from matplotlib import pyplot
from matplotlib.patches import PathPatch
from matplotlib.path import Path
from keras.models import Sequential
from keras.layers import LSTM
from keras.layers import Dense

# generate a rectangle with random width and height
def random_rectangle():
	width, height = random(), random()
	points = list()
	# bottom left
	points.append([0.0, 0.0])
	# bottom right
	points.append([width, 0.0])
	# top right
	points.append([width, height])
	# top left
	points.append([0.0, height])
	return points

# plot a rectangle
def plot_rectangle(rect):
	# close the rectangle path
	rect.append(rect[0])
	# define path
	codes = [Path.MOVETO, Path.LINETO, Path.LINETO, Path.LINETO, Path.CLOSEPOLY]
	path = Path(rect, codes)
	axis = pyplot.gca()
	patch = PathPatch(path)
	# add shape to plot
	axis.add_patch(patch)
	axis.set_xlim(-0.1,1.1)
	axis.set_ylim(-0.1,1.1)
	pyplot.show()

# generate input and output sequences for one random rectangle
def get_samples():
	# generate rectangle
	rect = random_rectangle()
	X, y = list(), list()
	# create input output pairs for each coordinate
	for i in range(1, len(rect)):
		X.append(rect[i-1])
		y.append(rect[i])
	# convert input sequence shape to have 1 time step and 2 features
	X, y = array(X), array(y)
	X = X.reshape((X.shape[0], 1, 2))
	return X, y

# use a fit LSTM model to generate a new rectangle from scratch
def generate_rectangle(model):
	rect = list()
	# use [0,0] to seed the generation process
	last = array([0.0,0.0]).reshape((1, 1, 2))
	rect.append([[y for y in x] for x in last[0]][0])
	# generate the remaining 3 coordinates
	for _ in range(3):
		# predict the next coordinate
		yhat = model.predict(last, verbose=0)
		# use this output as input for the next prediction
		last = yhat.reshape((1, 1, 2))
		# store coordinate
		rect.append([[y for y in x] for x in last[0]][0])
	return rect

# define model
model = Sequential()
model.add(LSTM(10, input_shape=(1, 2)))
model.add(Dense(2, activation='linear'))
model.compile(loss='mae', optimizer='adam')
model.summary()

# fit model
for i in range(25000):
	X, y = get_samples()
	model.fit(X, y, epochs=1, verbose=2, shuffle=False)

# generate new shapes from scratch
rect = generate_rectangle(model)
plot_rectangle(rect)

**Comment by Saduf2019** [comment]
@raposter 
Please share a colab gist with the error reported.

**Comment by raposter** [comment]
I've removed Version 3.9 and cannot regererate any errors and am unfamiliar with what a colab gist is.  So, I cannot help you with this problem.  The code I included is from a tutorial and can readily be run in the 3.9 environment.Thanks
    On Friday, April 9, 2021, 12:07:43 PM PDT, Saduf2019 ***@***.***> wrote:  
 
 


@raposter
Please share a colab gist with the error reported.

**Comment by Saduf2019** [comment]
@raposter 
Please provide the code in format such that we can replicate the error reported, as the code shared has too many indentation errors.

**Comment by raposter** [comment]
Enclosed is the source code taken from the LSTM Networks book by Jason Bownlee.Thanks
    On Wednesday, April 21, 2021, 12:05:11 PM PDT, Saduf2019 ***@***.***> wrote:  
 
 


@raposter
Please provide the code in format such that we can replicate the error reported, as the code shared has too many indentation errors.

**Comment by jvishnuvardhan** [comment]
@raposter I formatted your code and ran it in colab (which is similar to Jupyter notebook) and I don't see any issue there. Please check the [gist here](https://colab.research.google.com/gist/jvishnuvardhan/483b8bebb873c0ba894ab0fef496bbb9/untitled.ipynb). 

Two things i modified in your code
1. imported keras from tensorflow (`from tensorflow import keras`)
2. changed for loop iterations from 25000 to 250 (just to speed up)

Please verify it and close if the issue was resolved for you. If it was not resolved, please share a jupyter notebook or python file for reproducing the issue. Thanks

**Comment by raposter** [comment]
My issue occurred with Python 3.9.  It looks like from the gist, that Python 3.7 was used.  If not, and it works in Python 3.9 then please close the issue.Thanks  
    On Thursday, April 22, 2021, 02:38:09 PM PDT, Vishnuvardhan Janapati ***@***.***> wrote:  
 
 


@raposter I formatted your code and ran it in colab (which is similar to Jupyter notebook) and I don't see any issue there. Please check the gist here.

Two things i modified in your code
   
   - imported keras from tensorflow (from tensorflow import keras)
   - changed for loop iterations from 25000 to 250 (just to speed up)

Please verify it and close if the issue was resolved for you. If it was not resolved, please share a jupyter notebook or python file for reproducing the issue. Thanks

**Comment by jvishnuvardhan** [comment]
@raposter Is it possible for you to check with `python3.9`? You can download `*.py` file from my colab and run it. Thanks!
